Src/coffeescript/cql-execution/src/elm/quantity.coffee

```diff
@@ -28,16 +28,31 @@ module.exports.Quantity = class Quantity extends Expression
     if other instanceof Quantity and other.unit == @unit then @value > parseFloat other.value else null
 
   before: (other) ->
-    if other instanceof Quantity and other.unit == @unit then @value < parseFloat other.value else null
+    if other instanceof Quantity and other.unit == @unit 
+      @value < parseFloat other.value 
+    else if other instanceof Quantity and other.unit of time_units and @unit of time_units
+      thisSmallestDuration = smallestDuration(@)
+      otherSmallestDuration = smallestDuration(other)
+      thisSmallestDuration < otherSmallestDuration
+    else null
```

Comment:
it looks like this set of conversions should be done for all comparisons: `sameOrBefore`, `sameOrAfter`, `after`, `before`, `equals`. 

It looks like it should also be done for the operations, perhaps (division, subtraction, etc.). If you convert it for operations, it seems like you'll need to do a conversion back to some unit. Which unit? Maybe to the most granular unit? e.g. 5 hours - 30 minutes gives 270 minutes, 4 hours, or 4 hours 30 minutes? How will rounding be handled?